From e0afad1c93da103e4c3956b283402e7eedaf6e3d Mon Sep 17 00:00:00 2001
From: Mihai Serban <mihai.serban@nxp.com>
Date: Wed, 26 Jul 2017 11:17:09 +0300
Subject: [PATCH 2250/5242] MLK-16007-2: ASoC: fsl_asrc: Enable automatic
 configuration for P2P

commit  515e0c6d3fd73aba0ec1a3a91c4f53fa50682105 from
https://source.codeaurora.org/external/imx/linux-imx.git

Use automatic selection of processing options and internal measured
ratio for P2P conversions.

The conversion done by ASRC depends on the IPG master clock frequency
that can have any value between 130MHz and 200MHz. The documentation
states that to support 10 channels with 192KHz sampling rate the
master clock frequency must be at least 160MHz.
When the master clock cannot be configured to faster frequencies the
ASRC can still convert the samples but it have to be configured to
automatically adjust the processing options and conversion ratio.

Signed-off-by: Mihai Serban <mihai.serban@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_asrc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_asrc.c b/sound/soc/fsl/fsl_asrc.c
index e9d57bd..1f74b4f 100644
--- a/sound/soc/fsl/fsl_asrc.c
+++ b/sound/soc/fsl/fsl_asrc.c
@@ -381,7 +381,7 @@ static int fsl_asrc_config_pair(struct fsl_asrc_pair *pair, bool p2p_in, bool p2
 	 * When M2M mode, output rate should also need to align with the out
 	 * samplerate, but M2M must use less time to achieve good performance.
 	 */
-	if (p2p_out)
+	if (p2p_out || p2p_in)
 		div[OUT] = clk_get_rate(clk) / outrate;
 	else
 		div[OUT] = clk_get_rate(clk) / IDEAL_RATIO_RATE;
@@ -575,7 +575,7 @@ static int fsl_asrc_dai_hw_params(struct snd_pcm_substream *substream,
 
 	config.pair = pair->index;
 	config.channel_num = channels;
-	config.inclk = INCLK_NONE;
+	config.inclk = INCLK_ASRCK1_CLK;
 	config.outclk = OUTCLK_ASRCK1_CLK;
 
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK) {
-- 
1.7.9.5

